src/osm_configurator/view/states/main_window.py

```diff
@@ -32,16 +63,53 @@ def __init__(self, export_controller, category_controller, project_controller, s
             calculation_controller (calculation_controller.CalculationController): Respective controller.
             cut_out_controller (cut_out_controller.CutOutController): Respective controller.
             data_visualization_controller (data_visualization_controller.DataVisualizationController): Respective controller.
             osm_data_controller (osm_data_controller.OSMDataController): Respective controller.
         """
-        pass
+        # Creating the mainWindow and setting its position, and making it resizable
+        self._window = customtkinter.CTk()
+        self._window.title(main_window_constants_i.MainWindowConstants.WINDOW_TITLE.value)
+
+        # Selecting the primary Monitor to get accurate location for centering the window
+        primary_monitor = None
+        for monitor in screeninfo.get_monitors():
+            if monitor.is_primary:
+                primary_monitor = monitor
+                break
+
+        screen_height = primary_monitor.height
+        screen_width = primary_monitor.width
+
+        true_height = screen_height / 2 - main_window_constants_i.MainWindowConstants.MAIN_WINDOW_HEIGHT.value / 2
+        true_width = screen_width / 2 - main_window_constants_i.MainWindowConstants.MAIN_WINDOW_WIDTH.value / 2
+
+        self._window.geometry("%dx%d+%d+%d" % (main_window_constants_i.MainWindowConstants.MAIN_WINDOW_WIDTH.value,
+                                               main_window_constants_i.MainWindowConstants.MAIN_WINDOW_HEIGHT.value,
+                                               true_width, true_height))
+        self._window.minsize(main_window_constants_i.MainWindowConstants.MAIN_WINDOW_WIDTH_MINIMUM.value,
+                             main_window_constants_i.MainWindowConstants.MAIN_WINDOW_HEIGHT_MINIMUM.value)
+        self._window.resizable(True, True)
+
+        # Configurating the grid for the Application
+        self._window.grid_columnconfigure(0, weight=COLUM_WEIGHT)
+        self._window.grid_rowconfigure(0, weight=TOP_ROW_WEIGHT)
+        self._window.grid_rowconfigure(1, weight=MIDDLE_ROW_WEIGHT)
+        self._window.grid_rowconfigure(2, weight=BOTTOM_ROW_WEIGHT)
 
+        # Creating the StateManager
+        self._state_manager = state_manager_i.StateManager(self, export_controller, category_controller,
+                                                           project_controller,
+                                                           settings_controller, aggregation_controller,
+                                                           calculation_controller,
+                                                           cut_out_controller, data_visualization_controller,
+                                                           osm_data_controller)
 
-    def change_state(self, last_state, new_state):
+        # Starting the mainloop
+        self._window.mainloop()
```

Comment:
maybe add a method start_main_window() that enters the manloop. Currently it is a little bit weird, because the constructor of the class is never finished. With start_main_window() you could first us ethe constructor to create the window and later the start_main_window() method to start the loop

Reply:
start of loop outsourced to another method